wordpress-mobile/WordPress-Android
# Cleanup LocaleManager

This PR does two things:

* Cleanup `LocaleManager` by removing unused code and fixing warnings
* Convert `LocaleMananger` to Kotlin

These actions will make it easier to eventually do away with storing the app language in our shared preferences, enabling the system to be in charge of the language.

I left as much logic unchanged as possible, but I did need to make minor modifications to `createSortedLanguageDisplayStrings` to get it to work. This change can be tested two ways:

Test 1:

* My Site > Site Settings
* Tap Language
* Verify the language picker appears correctly

Test 2:

* On a pre-API 13 device, go to App Settings
* Tap the Interface Language
* Verify the language picker appears correctly
